### ISSUE TITLE
HR Zones are historicalish

### ISSUE THREAD

**Issue by mayfield** [SauceLLC/sauce4strava#22]
Strangely HR zones are tied to activities and they are FOR-EV-AR.   Updating your account settings with new zones only affects new activities.  You can never change HR zones from the past.  Well this is bad because if you accidentally have incorrect data in there you can't fix it in strava land (I just realized my zones were wrong for some unknown period of time).  

We can improve this for Sauce users by importing and then overriding the HR zones based on our own HRZonesHistory data like FTP and weight.

For athletes we sync, we should also find a way to override the hr zones being used in an activity, at least for our own hr based calcs like tTSS.

In short,

 * Make hrzones historical.
 * Modify the HR zones processor to check the HR zones of activities on a periodic scale of some sort.  Ie. Check one activity for every week/month to see if the zones have updated.  This is for initial import only as we'll want to make our history data authoritative.

**Comment by mayfield** [comment]
Closing this, as I've made a change that pulls HR Zones from latest activities.  I'm not 100% on it, but we have more pressing issues.
